packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.swift

```diff
@@ -0,0 +1,1547 @@
+protocol FLTCam: FlutterTexture, AVCaptureVideoDataOutputSampleBufferDelegate,
+  AVCaptureAudioDataOutputSampleBufferDelegate
+{
+  /// The API instance used to communicate with the Dart side of the plugin.
+  /// Once initially set, this should only ever be accessed on the main thread.
+  var dartAPI: FCPCameraEventApi? { get set }
+
+  var onFrameAvailable: (() -> Void)? { get set }
+
+  /// Format used for video and image streaming.
+  var videoFormat: FourCharCode { get set }
+
+  var isPreviewPaused: Bool { get }
+
+  var minimumAvailableZoomFactor: CGFloat { get }
+  var maximumAvailableZoomFactor: CGFloat { get }
+  var minimumExposureOffset: CGFloat { get }
+  var maximumExposureOffset: CGFloat { get }
+
+  func setUpCaptureSessionForAudioIfNeeded()
+
+  func reportInitializationState()
+
+  /// Acknowledges the receipt of one image stream frame.
+  func receivedImageStreamData()
+
+  func start()
+  func stop()
+
+  /// Starts recording a video with an optional streaming messenger.
+  func startVideoRecording(
+    completion: @escaping (_ error: FlutterError?) -> Void,
+    messengerForStreaming: FlutterBinaryMessenger?
+  )
+  func pauseVideoRecording()
+  func resumeVideoRecording()
+  func stopVideoRecording(completion: @escaping (_ path: String?, _ error: FlutterError?) -> Void)
+
+  func captureToFile(completion: @escaping (_ path: String?, _ error: FlutterError?) -> Void)
+
+  func setDeviceOrientation(_ orientation: UIDeviceOrientation)
+  func lockCaptureOrientation(_ orientation: FCPPlatformDeviceOrientation)
+  func unlockCaptureOrientation()
+
+  func setImageFileFormat(_ fileFormat: FCPPlatformImageFileFormat)
+
+  func setExposureMode(_ mode: FCPPlatformExposureMode)
+  func setExposureOffset(_ offset: Double)
+  func setExposurePoint(
+    _ point: FCPPlatformPoint?,
+    withCompletion: @escaping (_ error: FlutterError?) -> Void
+  )
+
+  func setFocusMode(_ mode: FCPPlatformFocusMode)
+  func setFocusPoint(
+    _ point: FCPPlatformPoint?,
+    completion: @escaping (_ error: FlutterError?) -> Void
+  )
+
+  func setZoomLevel(_ zoom: CGFloat, withCompletion: @escaping (_ error: FlutterError?) -> Void)
+
+  func setFlashMode(
+    _ mode: FCPPlatformFlashMode,
+    withCompletion: @escaping (_ error: FlutterError?) -> Void
+  )
+
+  func pausePreview()
+  func resumePreview()
+
+  func setDescriptionWhileRecording(
+    _ cameraName: String,
+    withCompletion: @escaping (_ error: FlutterError?) -> Void
+  )
+
+  func startImageStream(with: FlutterBinaryMessenger)
+  func stopImageStream()
+
+  // Override to make `AVCaptureVideoDataOutputSampleBufferDelegate`/
+  // `AVCaptureAudioDataOutputSampleBufferDelegate` method non optional
+  override func captureOutput(
+    _ output: AVCaptureOutput,
+    didOutput sampleBuffer: CMSampleBuffer,
+    from connection: AVCaptureConnection
+  )
+
+  func close()
+}
+
+class FLTDefaultCam: NSObject, FLTCam {
```

Comment:
nit: final class. 

this file is getting very long. can you move it to a separate file?